src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -138,6 +138,7 @@ export class DatasetService {
   }
 
   private cacheDataStructure(dataType: string, dataStructure: DataStructure) {
+    this.setLastCacheTime(Date.now());
```

Comment:
is it just a single timestamp for all caches or can different data structures have different cache time?

Reply:
It's just a single timestamp for all caches for now. Currently, we haven't had any requirements to make it per data structure/data type.